### PR TITLE
feat(JOML): migrate world renderer

### PR DESCRIPTION
**Pull request by pollend**

### CODE REVIEW

File: engine/src/main/java/org/terasology/rendering/world/RenderableWorldImpl.java [hunk 1]
```diff
@@ -196,33 +196,31 @@ public void update() {
...
-                if (chunk != null) {
-                    chunksInProximityOfCamera.add(chunk);
-                    chunksHaveBeenAdded = true;
+            for (Vector3ic toAdd : newRenderableRegion) {
+                if (!renderableRegion.contains(toAdd)) {
+                    chunk = chunkProvider.getChunk(JomlUtil.from(toAdd));
+                    if (chunk != null) {
+                        chunksInProximityOfCamera.add(chunk);
+                        chunksHaveBeenAdded = true;
+                    }
```
Comment (pollend) [inline engine/src/main/java/org/terasology/rendering/world/RenderableWorldImpl.java:223]: Hopefully this is migrated correctly.

File: engine/src/main/java/org/terasology/rendering/world/RenderableWorldImpl.java [hunk 2]
```diff
@@ -196,33 +196,31 @@ public void update() {
...
-
             boolean chunksHaveBeenAdded = false;
-            Iterator<Vector3i> chunksToAdd = newRenderableRegion.subtract(renderableRegion);
-            while (chunksToAdd.hasNext()) {
-                chunkPosition = chunksToAdd.next();
-                chunk = chunkProvider.getChunk(chunkPosition);
-                if (chunk != null) {
-                    chunksInProximityOfCamera.add(chunk);
-                    chunksHaveBeenAdded = true;
+            for (Vector3ic toAdd : newRenderableRegion) {
```
Comment (jdrueckert) [inline engine/src/main/java/org/terasology/rendering/world/RenderableWorldImpl.java:217]:
```suggestion
            for (Vector3ic chunkPositionToAdd : newRenderableRegion) {
```

File: engine/src/main/java/org/terasology/rendering/world/RenderableWorldImpl.java (hunk 1 above)
Comment (jdrueckert) [reply engine/src/main/java/org/terasology/rendering/world/RenderableWorldImpl.java:223]: I think you did, yes.

File: engine/src/main/java/org/terasology/rendering/world/RenderableWorldImpl.java [hunk 3]
```diff
@@ -196,33 +196,31 @@ public void update() {
      * @return True if the list was changed
      */
     @Override
-    public boolean updateChunksInProximity(Region3i newRenderableRegion) {
+    public boolean updateChunksInProximity(BlockRegion newRenderableRegion) {
         if (!newRenderableRegion.equals(renderableRegion)) {
-            Vector3i chunkPosition;
             RenderableChunk chunk;
+            for (Vector3ic toRemove : renderableRegion) {
```
Comment (jdrueckert) [inline engine/src/main/java/org/terasology/rendering/world/RenderableWorldImpl.java:202]:
```suggestion
            for (Vector3ic chunkPositionToRemove : renderableRegion) {
```

**Review by jdrueckert — CHANGES_REQUESTED** [review]

**Review by jdrueckert — APPROVED** [review]

**Pull request merged**